syntax = "proto3";

package cod_pb;

message Value {
    oneof Value {
        int32 value_int = 1;
        float value_float = 2;
        string value_string = 3;    
    }
}

message ArrayValue {
    repeated Value value = 1;
}

message Variable {
    string name = 1;
    oneof Value {
        int32 value_int = 2;
        float value_float = 3;
        string value_string = 4;   
        ArrayValue value_array = 5;
    }
}

message Variables {
    repeated Variable variable = 1;
}

message Object {
    string name = 1;
    Variables variables = 2;
    repeated Object objects = 3;
}

message Objects {
    repeated Object object = 1;
}
